core/src/wsv.rs

```diff
@@ -54,7 +55,37 @@ pub struct World {
     /// Triggers
     pub(crate) triggers: TriggerSet,
     /// Chain of *runtime* validators
-    pub(crate) validator: crate::validator::Validator,
+    pub(crate) validator: RwLock<Option<crate::validator::Validator>>,
```

Comment:
as far as I am aware we will not need internal syncing inside `wsv`. The whole `wsv` will be guarded with a sync primitive

Reply:
Looks like a duplicate of the comment above, so I'll close this